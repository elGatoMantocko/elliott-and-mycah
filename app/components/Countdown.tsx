import Box from '@material-ui/core/Box';
import Card from '@material-ui/core/Card';
import Hidden from '@material-ui/core/Hidden';
import Typography from '@material-ui/core/Typography';
import * as moment from 'moment';
import * as React from 'react';
import { useEffect, useState } from 'react';

type CounterCardProps = Readonly<{ value: number; unit: string }>;
const CounterCard = ({ value }: CounterCardProps) => (
  <Card raised>
    <Box margin="0.5rem">
      <Typography variant="h1">
        {value < 10 ? '0' : ''}
        {value}
      </Typography>
    </Box>
  </Card>
);

type CountdownProps = Readonly<{ date: Date }>;
export const Countdown = ({ date }: CountdownProps) => {
  const [time, setTime] = useState(new Date());

  useEffect(() => {
    setInterval(() => setTime(new Date()), 1000);
  }, [time, setTime]);

  const momentDate = moment(date);

  const days = momentDate.diff(time, 'days');
  const daysNormalized = momentDate.subtract(days, 'days');

  const hours = daysNormalized.diff(time, 'hours');
  const hoursNormalized = daysNormalized.subtract(hours, 'hours');

  const minutes = hoursNormalized.diff(time, 'minutes');
  const minutesNormalized = hoursNormalized.subtract(minutes, 'minutes');

  const seconds = minutesNormalized.diff(time, 'seconds');

  return (
    <Box display="flex" justifyContent="space-around" width="100%">
      <CounterCard value={days} unit="days" />
      <Hidden xsDown>
        <CounterCard value={hours} unit="hours" />
        <CounterCard value={minutes} unit="minutes" />
        <Hidden smDown>
          <CounterCard value={seconds} unit="seconds" />
        </Hidden>
      </Hidden>
    </Box>
  );
};
