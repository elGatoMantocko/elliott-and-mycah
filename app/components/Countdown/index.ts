export * from './Countdown';
export * from './useCountdown';
