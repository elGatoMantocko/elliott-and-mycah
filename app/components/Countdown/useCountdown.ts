import { intervalToDuration } from 'date-fns';
import { useEffect, useState } from 'react';

interface CountdownDeltas {
  years?: number;
  months?: number;
  days?: number;
}

export const useCountdown = (toDate: Date): CountdownDeltas => {
  const [duration, setDuration] = useState<Duration>(
    intervalToDuration({ start: toDate, end: new Date() }),
  );

  useEffect(() => {
    const id = setInterval(() => {
      const duration = intervalToDuration({ start: toDate, end: new Date() });
      setDuration(duration);
    }, 1000);
    return () => clearInterval(id);
  }, [setDuration, toDate]);

  return duration;
};
