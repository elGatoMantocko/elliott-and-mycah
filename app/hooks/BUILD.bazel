
load("@npm//@bazel/typescript:index.bzl", "ts_library")
load("@npm//eslint:index.bzl", "eslint")

filegroup(
    name = "source_files",
    srcs = glob(["*.tsx"]),
)

eslint(
    name = "eslint",
    data = [
        "//:.eslintrc",
        ":source_files",
        "@npm//:node_modules",
    ],
    args = ["**/*.tsx"],
)

ts_library(
    name = "ts_library",
    srcs = [":source_files"],
    deps = [
        "@npm//@material-ui/core",
        "@npm//@types",
        "@npm//react",
    ],
    visibility = ["//visibility:public"],
)