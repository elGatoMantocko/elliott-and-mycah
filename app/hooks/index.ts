export * from './result';
export * from './scrollToTop';
export * from './serviceWorker';
export * from './theme';
