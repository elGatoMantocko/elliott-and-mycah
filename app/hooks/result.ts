import { DependencyList, useCallback, useEffect, useState } from 'react';

import { State, state } from '../models';

export enum ResultState {
  NotStarted = 'not-started',
  Pending = 'pending',
  Value = 'value',
  Error = 'error',
}

export type Result<T, U> =
  | State<ResultState.NotStarted>
  | State<ResultState.Pending>
  | State<ResultState.Value, T>
  | State<ResultState.Error, U>;

const PENDING: State<ResultState.Pending> = state(ResultState.Pending);
const NOT_STARTED: State<ResultState.NotStarted> = state(ResultState.NotStarted);

/**
 * Turns an unknown error into an {@link Error}
 * @param err unknown error
 * @returns a valid error object
 */
function handleUnknownError(err: unknown): Error {
  if (err instanceof Error) {
    return err;
  } else if (typeof err === 'string') {
    return new Error(err);
  } else if (typeof err === 'object') {
    return new Error(`encountered error in useCallableResult: ${JSON.stringify(err)}`);
  } else {
    return new Error('encountered unknown error in useCallableResult');
  }
}

/**
 * Function to get a result from an asynchronous function.
 * @param request asynchronous function to map to a result state
 * @param deps list of deps used in the request
 * @returns result state object
 */
export const useResult = <C extends () => Promise<unknown>, CValue extends Awaited<ReturnType<C>>>(
  request: C extends () => Promise<CValue> ? C : never,
  deps: DependencyList,
) => {
  const [result, setResult] = useState<Result<CValue, Error>>(NOT_STARTED);

  useEffect(() => {
    let isSubscribed = true;

    // only set state and make requests as long as this hook is still "mounted" to the dom
    // this is mostly just so we don't freak the tests out when things get unmounted
    if (isSubscribed) {
      setResult(PENDING);
      request()
        .then((data) => setResult(state(ResultState.Value, data)))
        .catch((err: unknown) => setResult(state(ResultState.Error, handleUnknownError(err))));
    }

    // if the `useResult` hook is torn down, we want to make sure we don't have any
    // lingering calls being made
    return () => {
      isSubscribed = false;
    };
    // The caller is defining the dependencies to this so we intentionally
    // want to ignore the exhaustive deps rule here.
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [...deps]);
  return result;
};

/**
 * Hook to get a callable function that sets the result state. Can be reset to `not-started`
 * with the 3rd option in the tuple.
 * @param request function to call
 * @returns tuple representing the `[result-state, callable, reset]` options
 */
export const useCallableResult = <CArgs extends unknown[], CResponse>(
  request: (...args: CArgs) => Promise<CResponse>,
): [Result<CResponse, Error>, (...args: CArgs) => void, () => void] => {
  const [result, setResult] = useState<Result<CResponse, Error>>(NOT_STARTED);

  const call = useCallback(
    async (...args: CArgs) => {
      setResult(PENDING);

      try {
        const t = await request(...args);

        setResult(state(ResultState.Value, t));
      } catch (err) {
        setResult(state(ResultState.Error, handleUnknownError(err)));
      }
    },
    [request],
  );

  return [result, call, () => setResult(NOT_STARTED)];
};
