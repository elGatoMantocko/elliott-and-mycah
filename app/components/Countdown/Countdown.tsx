import { Box } from '@mui/material';
import React from 'react';

import { ScriptTypography } from '../ScriptTypography';
import { useCountdown } from './useCountdown';

type Units = 'years' | 'months' | 'days';
type EnUnits = { [Unit in Units]: (isZeroOrMoreThanOne: boolean) => string };

const enUnits: EnUnits = {
  years: (check) => (check ? 'years' : 'year'),
  months: (check) => (check ? 'months' : 'month'),
  days: (check) => (check ? 'days' : 'day'),
};

const pluralize = (amount: number, key: Units): string => enUnits[key](amount === 0 || amount > 1);

type CounterCardProps = Readonly<{ value: number | undefined; unit: Units }>;
const CounterCard = ({ value, unit }: CounterCardProps) => (
  <>
    {value != null && value !== 0 && (
      <Box textAlign="center" margin="0.5rem">
        <ScriptTypography data-testid={`${unit}-delta`} variant="h1">
          {value}
        </ScriptTypography>
        <ScriptTypography variant="h4">{pluralize(value, unit)}</ScriptTypography>
      </Box>
    )}
  </>
);

type CountdownProps = Readonly<{ toDate: Date }>;
export const Countdown = ({ toDate }: CountdownProps) => {
  const { years, months, days } = useCountdown(toDate);

  return (
    <Box display="flex" justifyContent="space-around" width="100%">
      <CounterCard value={years} unit="years" />
      <CounterCard value={months} unit="months" />
      <CounterCard value={days} unit="days" />
    </Box>
  );
};
