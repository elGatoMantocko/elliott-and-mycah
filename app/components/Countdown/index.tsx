import { Box, Hidden } from '@mui/material';
import React from 'react';

import { ScriptTypography } from '../ScriptTypography';
import { useCountdown } from './useCountdown';

type CounterCardProps = Readonly<{ value: number; unit: string }>;
const CounterCard = ({ value, unit }: CounterCardProps) => (
  <Box textAlign="center" margin="0.5rem">
    <ScriptTypography variant="h1">
      {value < 10 ? '0' : ''}
      {value}
    </ScriptTypography>
    <ScriptTypography variant="h4">{unit}</ScriptTypography>
  </Box>
);
type CountdownProps = Readonly<{ toDate: Date }>;
export const Countdown = ({ toDate }: CountdownProps) => {
  const { days, hours, minutes, seconds } = useCountdown(toDate);

  return (
    <Box display="flex" justifyContent="space-around" width="100%">
      <CounterCard value={days} unit="days" />
      <Hidden smDown>
        <CounterCard value={hours} unit="hours" />
        <CounterCard value={minutes} unit="minutes" />
        <Hidden mdDown>
          <CounterCard value={seconds} unit="seconds" />
        </Hidden>
      </Hidden>
    </Box>
  );
};
