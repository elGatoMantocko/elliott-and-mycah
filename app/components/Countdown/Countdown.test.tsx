import { ThemeProvider } from '@mui/material/styles';
import { render, screen } from '@testing-library/react';
import { sub } from 'date-fns';
import React, { PropsWithChildren } from 'react';

import { useCustomTheme } from '../../hooks';
import { Countdown } from './Countdown';

const MockProviders = ({ children }: PropsWithChildren<unknown>) => {
  const theme = useCustomTheme();
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

it('should render', async () => {
  render(
    <div data-testid="basic-countdown">
      <Countdown toDate={sub(new Date(), { years: 1, months: 1, days: 1 })} />,
    </div>,
    { wrapper: MockProviders },
  );

  expect(screen.getByTestId('basic-countdown')).toMatchSnapshot('single-date');
});

it('should render plural duration', async () => {
  render(
    <div data-testid="multiple-countdown">
      <Countdown toDate={sub(new Date(), { years: 2, months: 2, days: 2 })} />,
    </div>,
    { wrapper: MockProviders },
  );

  expect(screen.getByTestId('multiple-countdown')).toMatchSnapshot('plural-date');
});

it('should not render null (0) deltas', async () => {
  render(
    <div data-testid="null-countdown">
      <Countdown toDate={sub(new Date(), { years: 1 })} />,
    </div>,
    { wrapper: MockProviders },
  );

  expect(screen.getByTestId('null-countdown')).toMatchSnapshot('null-like-deltas');
});
