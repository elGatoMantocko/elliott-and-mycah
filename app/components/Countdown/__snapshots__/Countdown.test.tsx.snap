// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`should render countdown 1 year : 1 month : 1 day for screen-size 764 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          year
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          month
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          day
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 1 year : 1 month : 1 day for screen-size 1024 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          year
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          month
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          day
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 1 year : 1 month : undefined day for screen-size 764 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          year
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          month
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 1 year : 1 month : undefined day for screen-size 1024 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          year
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          month
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 1 year : undefined month : undefined day for screen-size 764 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          year
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 1 year : undefined month : undefined day for screen-size 1024 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          1
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          year
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 2 year : 2 month : 2 day for screen-size 764 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          2
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          years
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          2
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          months
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          2
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          days
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 2 year : 2 month : 2 day for screen-size 1024 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          2
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          years
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          2
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          months
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          2
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          days
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 3 year : 3 month : 3 day for screen-size 764 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          3
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          years
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          3
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          months
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          3
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          days
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`should render countdown 3 year : 3 month : 3 day for screen-size 1024 1`] = `
<DocumentFragment>
  <div
    class="MuiBox-root css-kr4k96"
  >
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          3
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          years
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          3
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          months
        </h4>
      </div>
    </div>
    <div
      class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation5 css-30kr64-MuiPaper-root"
    >
      <div
        class="MuiBox-root css-1erotlu"
      >
        <h1
          class="MuiTypography-root MuiTypography-h1 css-tatl18-MuiTypography-root"
        >
          3
        </h1>
        <h4
          class="MuiTypography-root MuiTypography-h4 css-g8o2ax-MuiTypography-root"
        >
          days
        </h4>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
